docs/docs/docs/01-core/configuration/00-intro.md

```diff
@@ -0,0 +1,115 @@
+---
+title: Introduction
+tags:
+  - configuration
+---
+
+# Configuration
+
+This section is an overview of how configuration files are loaded in Strapi and how they should be used.
+
+## Config loading
+
+Loading of configuration is the very first thing Strapi does in bootstrap (after setting signal handlers), which is necessary because most of the Strapi bootstrap process depends on values from the configuration.
+
+This means that configuration loading happens even before Strapi creates a logger. Therefore any errors or warnings from the configuration unfortunately must output to `console`
+
+NOTE: For now. in theory, we could define some configuration files that do not depend on being passed a `strapi` object and could provide logger settings in there, but at this time we do not have that capability.
+
+Currently, Strapi loads every .js and .json file within a project's `./config` directory directly into the strapi config object. That is, if there is a file called `./config/myconfig.js` it will be loaded automatically and accessible from `strapi.config.get('myconfig')`
+
+(NOT YET IMPLEMENTED) Starting in Strapi v5, will load every defined value from the `STRAPI_` environment namespace into the configuration, using underscore (\_) as the object delimiter. For example, if you define `STRAPI_ADMIN_APP_KEYS=abcd,efg` it will be available from `strapi.config.get('admin.app.keys')` as `['abcd', 'efg']`
```

Comment:
Since this is going to V5, I would either keep track of this on Jira (or in a TODO somewhere) or remove the not yet implemented